docpages/example_programs/the_basics/using_callback_functions.md

```diff
@@ -1,70 +1,66 @@
+    dpp::cluster bot(BOT_TOKEN, dpp::i_default_intents | dpp::i_message_content);
+    // the second argument is a bitmask of intents - i_message_content is needed to get messages
 
     bot.on_log(dpp::utility::cout_logger());
 
-    bot.on_slashcommand([&bot, gm](const dpp::slashcommand_t& event) -> void {
-        if (event.command.get_command_name() == "msg-get") {
+    // The event is fired when someone issues your commands
+    bot.on_slashcommand([&bot](const dpp::slashcommand_t& event) -> void {
+        if (event.command.get_command_name() == "msgs-get") {
+            int64_t limit = std::get<int64_t>(event.get_parameter("quantity"));
 
-            // get a message using its own and channel's ID
-            bot.message_get(gm.message_id, gm.channel_id, [&bot, event](const dpp::confirmation_callback_t& callback) -> void {
+            // get messages using ID of the channel the command was issued in
+            bot.messages_get(event.command.channel_id, 0, 0, 0, limit, [event](const dpp::confirmation_callback_t& callback) -> void {
                 if (callback.is_error()) { // catching an error to log it
-                    bot.log(dpp::loglevel::ll_error, callback.get_error().message);
+                    std::cout << callback.get_error().message << std::endl;
                     return;
                 }
 
-                // callback.value will contain the message we are getting but the type will be std::variant, so we will have to get the message itself.
-                auto message = callback.get <dpp::message>();
-                //std::get <dpp::message>(callback.value) would give the same result
+                auto messages = callback.get<dpp::message_map>();
+                // std::get <dpp::message_map>(callback.value) would give the same result
+                
+                std::string contents;
+                for (const auto& x : messages) { // here we iterate through the dpp::message_map we got from callback...
+                    contents += x.second.content + '\n'; // ...where x.first is ID of the current message and x.second is the message itself.
+                }
 
-                event.reply(message); // replies with the same message
+                event.reply(contents); // we will see all those messages we got, united as one!
             });
         }
-        if (event.command.get_command_name() == "channel-create") {
+        else if (event.command.get_command_name() == "channel-create") {
 
-            //create a text channel
+            // create a text channel
             dpp::channel channel = dpp::channel()
                 .set_name("test")
-                .set_type(dpp::channel_type::CHANNEL_TEXT)
-                .set_guild_id(1112789761032200306)
-                .set_parent_id(0);
+                .set_guild_id(event.command.guild_id);
+            // by default, the channel we are creating is a text one without a parent (category)
```

Comment:
I think you can remove that comment, I think it's clear the channel doesn't need a parent as it's a channel, not a thread

Reply:
yes master

Reply:
comment is no mor